src/js/components/ServiceOptions.js

```diff
@@ -0,0 +1,115 @@
+import classNames from 'classnames';
+import mixin from 'reactjs-mixin';
+import React from 'react';
+import {StoreMixin} from 'mesosphere-shared-reactjs';
+
+import CosmosPackagesStore from '../stores/CosmosPackagesStore';
+import Framework from '../structs/Framework';
+import InternalStorageMixin from '../mixins/InternalStorageMixin';
+import Service from '../structs/Service';
+
+const METHODS_TO_BIND = [
+  'handleEditConfigClick',
+  'handleConfigModalClose'
+];
+
+class ServiceOptions extends mixin(InternalStorageMixin, StoreMixin) {
+  constructor() {
+    super(...arguments);
+
+    this.state = {
+      editConfigModalOpen: false
+    };
+
+    this.internalStorage_set({
+      servicePackage: null
+    });
+
+    this.store_listeners = [{
+      name: 'cosmosPackages',
+      events: [
+        'descriptionError',
+        'descriptionSuccess'
+      ]
+    }];
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  componentDidMount() {
+    super.componentDidMount(...arguments);
+
+    if (this.props.service instanceof Framework) {
+      let serviceMetadata = this.props.service.getMetadata();
+      let {name, version} = serviceMetadata;
+
+      CosmosPackagesStore.fetchPackageDescription(name, version);
+    }
+  }
+
+  onCosmosPackagesStoreDescriptionError() {
+    this.internalStorage_update({servicePackage: null});
+    this.forceUpdate();
+  }
+
+  onCosmosPackagesStoreDescriptionSuccess() {
+    this.internalStorage_update({
+      servicePackage: CosmosPackagesStore.getPackageDetails()
+    });
+    this.forceUpdate();
+  }
+
+  handleEditConfigClick() {
+    let {servicePackage} = this.internalStorage_get();
+
+    if (servicePackage != null) {
+      this.setState({editConfigModalOpen: true});
+    }
+  }
+
+  handleConfigModalClose() {
+    this.setState({editConfigModalOpen: false});
+  }
+
+  render() {
+    let {service} = this.props;
+    let {servicePackage} = this.internalStorage_get();
+
+    let editButtonClasses = classNames('button button-inverse button-stroke', {
+      'disabled': servicePackage == null
+    });
+
+    let buttons = [
+      <a className={editButtonClasses} key="edit-config"
+        onClick={this.handleEditConfigClick}>
+        Edit
+      </a>
+    ];
+
+    if (service.getWebURL()) {
```

Comment:
You're right, for some reason I was thinking that the method was on the `Service` struct (and maybe it should be?) because `Framework` extends `Service`...